File: javaexamples/src/main/java/com/inappstory/javaexamples/notification/NotificationSubscribeSample.java
package com.inappstory.javaexamples.notification;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.inappstory.javaexamples.R;
import com.inappstory.sdk.AppearanceManager;
import com.inappstory.sdk.InAppStoryManager;
import com.inappstory.sdk.stories.outercallbacks.common.reader.StoryData;
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListCallback;
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListCallbackAdapter;
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListScrollCallback;
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListScrollCallbackAdapter;
import com.inappstory.sdk.stories.ui.list.ShownStoriesListItem;
import com.inappstory.sdk.stories.ui.list.StoriesList;

import java.util.List;

public class NotificationSubscribeSample extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_base_list);
        showStories();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    private void showStories() {
        StoriesList storiesList = findViewById(R.id.stories_list);
        storiesList.setAppearanceManager(new AppearanceManager());
        InAppStoryManager.getInstance().setShowStoryCallback((storyData, showStoryAction) -> {

        });
        InAppStoryManager.getInstance().setCloseStoryCallback((slideData, closeReader) -> {

        });
        boolean adapterCallback = false;
        if (adapterCallback) {
            storiesList.setCallback(new ListCallbackAdapter() {
                @Override
                public void storiesLoaded(int size, String feed) {

                }
            });
            storiesList.setScrollCallback(new ListScrollCallbackAdapter() {
                @Override
                public void onVisibleAreaUpdated(List<ShownStoriesListItem> list) {

                }
            });
        } else {
            storiesList.setCallback(new ListCallback() {
                @Override
                public void storiesLoaded(int size, String feed) {

                }

                @Override
                public void storiesUpdated(int i, String s) {

                }

                @Override
                public void loadError(String feed) {

                }

                @Override
                public void itemClick(StoryData storyData, int i) {

                }
            });
            storiesList.setScrollCallback(new ListScrollCallback() {
                @Override
                public void scrollStart() {

                }

                @Override
                public void onVisibleAreaUpdated(List<ShownStoriesListItem> list) {

                }

                @Override
                public void scrollEnd() {

                }
            });
        }
        storiesList.loadStories();
    }
}

File: kotlinexamples/src/main/java/com/inappstory/kotlinexamples/notification/NotificationSubscribeSample.kt
package com.inappstory.kotlinexamples.notification

import android.os.Bundle
import android.view.View
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.inappstory.kotlinexamples.R
import com.inappstory.kotlinexamples.utils.ListShimmerView
import com.inappstory.sdk.stories.ui.list.StoriesList
import com.inappstory.sdk.AppearanceManager
import com.inappstory.sdk.InAppStoryManager
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListCallbackAdapter
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListCallback
import com.inappstory.sdk.stories.outercallbacks.common.reader.*
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListScrollCallback
import com.inappstory.sdk.stories.outercallbacks.storieslist.ListScrollCallbackAdapter
import com.inappstory.sdk.stories.ui.list.ShownStoriesListItem
import com.inappstory.sdk.stories.utils.Sizes
import kotlinx.coroutines.*

class NotificationSubscribeSample : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shimmer_list)
        showStories()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    private fun showStories() {
        val storiesList = findViewById<StoriesList>(R.id.stories_list)
        val shimmer = findViewById<ListShimmerView>(R.id.shimmer)
        val shimmerLayout = findViewById<FrameLayout>(R.id.shimmerLayout)
        shimmer.imageWidth = Sizes.dpToPxExt(120).toFloat()
        shimmerLayout.visibility = View.VISIBLE
        storiesList.appearanceManager = AppearanceManager()
        InAppStoryManager.getInstance()
            .setShowStoryCallback { storyData, action -> }
        InAppStoryManager.getInstance()
            .setCloseStoryCallback { storyData, closeReader -> }
        val adapterCallback = false
        if (adapterCallback) {
            storiesList.setCallback(object : ListCallbackAdapter() {
                override fun storiesLoaded(size: Int,
                                           feed: String?,
                                           storyData: MutableList<StoryData>?
                ) {
                    hideShimmer(shimmerLayout)
                }

                override fun loadError(feed: String) {
                    hideShimmer(shimmerLayout)
                }

            })
            storiesList.setScrollCallback(object : ListScrollCallbackAdapter() {
                override fun onVisibleAreaUpdated(items: MutableList<ShownStoriesListItem>?) {

                }
            })
        } else {
            storiesList.setCallback(object : ListCallback {
                override fun storiesLoaded(size: Int, feed: String?, p2: MutableList<StoryData>?) {

                    hideShimmer(shimmerLayout)
                }

                override fun storiesUpdated(size: Int, feed: String?, p2: MutableList<StoryData>?) {

                }

                override fun loadError(feed: String) {
                    hideShimmer(shimmerLayout)
                }

                override fun itemClick(
                    storyData: StoryData?,
                    index: Int
                ) {
                }
            })
            storiesList.setScrollCallback(object : ListScrollCallback {
                override fun scrollStart() {

                }

                override fun onVisibleAreaUpdated(items: MutableList<ShownStoriesListItem>?) {

                }

                override fun scrollEnd() {

                }
            })
        }
        storiesList.loadStories()
    }

    fun hideShimmer(layout: FrameLayout) {
        lifecycleScope.launch {
            delay(500)
            withContext(Dispatchers.Main) {
                layout.visibility = View.GONE
            }
        }
    }
}